apps/arweave/src/ar_node_worker.erl

```diff
@@ -1023,33 +1012,61 @@ may_be_get_double_signing_proof2(Iterator, RootHash, LockedRewards, Height) ->
 		none ->
 			undefined;
 		{Addr, {_Timestamp, Proof2}, Iterator2} ->
-			{Key, Sig1, _CDiff1, _PrevCDiff1, _Preimage1,
-					Sig2, _CDiff2, _PrevCDiff2, _Preimage2} = Proof2,
+			{Pub, Sig1, CDiff1, PrevCDiff1, Preimage1,
+					Sig2, CDiff2, PrevCDiff2, Preimage2} = Proof2,
 			?LOG_INFO([{event, evaluating_double_signing_proof},
-				{key_size, byte_size(Key)},
+				{key_size, byte_size(Pub)},
 				{sig1_size, byte_size(Sig1)},
 				{sig2_size, byte_size(Sig2)},
 				{height, Height}]),
 			CheckKeyType =
-				case {byte_size(Key) == ?ECDSA_PUB_KEY_SIZE, Height >= ar_fork:height_2_9()} of
+				case {byte_size(Pub) == ?ECDSA_PUB_KEY_SIZE, Height >= ar_fork:height_2_9()} of
 					{true, false} ->
 						false;
 					{true, true} ->
 						byte_size(Sig1) == ?ECDSA_SIG_SIZE
 							andalso byte_size(Sig2) == ?ECDSA_SIG_SIZE;
 					_ ->
-						byte_size(Key) == ?RSA_BLOCK_SIG_SIZE
+						byte_size(Pub) == ?RSA_BLOCK_SIG_SIZE
 							andalso byte_size(Sig1) == ?RSA_BLOCK_SIG_SIZE
 							andalso byte_size(Sig2) == ?RSA_BLOCK_SIG_SIZE
 				end,
-			HasLockedReward =
+			CheckDifferentSignatures =
 				case CheckKeyType of
+					false ->
+						false;
+					true ->
+						Sig1 /= Sig2
```

Comment:
This is just a sanity check right? It shouldn't ever happen? i.e. we'd expect the signature to always be different unless the same block is selected twice?

Reply:
yes